From 603633fbeed3bb97740c7410c671eddc06b5713e Mon Sep 17 00:00:00 2001
From: Meng Li <Meng.Li@windriver.com>
Date: Mon, 29 Aug 2022 13:58:25 +0800
Subject: [PATCH] secboot: move bl2 base address to 0x34100000 when secure boot
 is enabled

When building atf code, the LOAD_ADDRESS is calculated as below:
LOAD_ADDRESS = BL2_BASE - FIP_HDR_SIZE - DTB_SIZE.
After nxp-sdk rc5 release, the dtb is moved to into atf code.
When secure boot is enabled, the public key used to verify kernel
is saved in dtb file and cause dtb becomes bigger. In this way,
the LOAD_ADDRESS becomes lower than before, and enters the
address range 0x34008000~0x34078000 that is used by BootROM
for internal operation. So, move BL2_BASE to another address so
that there is no conflict with reversed sram memory even if dtb
file is extended when secure boot feature is enabed.

Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 plat/nxp/s32/s32cc/s32_common.mk | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/plat/nxp/s32/s32cc/s32_common.mk b/plat/nxp/s32/s32cc/s32_common.mk
index b43571240..adef884fa 100644
--- a/plat/nxp/s32/s32cc/s32_common.mk
+++ b/plat/nxp/s32/s32cc/s32_common.mk
@@ -430,7 +430,7 @@ ${MKIMAGE_FIP_CONF_FILE}: ${BL2_W_DTB_SIZE_FILE} FORCE
	echo "DATA_FILE SIZE $$BL2_W_DTB_SIZE" >> $@
 
 ifneq (${HSE_SUPPORT},)
-BL2_BASE		?= 0x34085000
+BL2_BASE		?= 0x34100000
 else
 BL2_BASE		?= 0x34302000
 endif
-- 
2.34.1

